### PR TITLE
OCT-1842: Reduce sentry events flood

### PR DESCRIPTION
**Pull request by mike-code**
## Description

Since our app is structured in a way that exceptions are expected, then caught and displayed as valid http responses, sentry must not report those exceptions as issues. It should thus not report "handled" exceptions.

## Definition of Done

1. [ ] Acceptance criteria are met.
2. [ ] PR is manually tested before the merge by developer(s).
    - [ ] Happy path is manually checked.
3. [ ] PR is manually tested by QA when their assistance is required (1).
    - [ ] Octant Areas & Test Cases are checked for impact and updated if required (2).
4. [ ] Unit tests are added unless there is a reason to omit them.
5. [ ] Automated tests are added when required.
6. [ ] The code is merged.
7. [ ] Tech documentation is added / updated, reviewed and approved (including mandatory approval by a code owner, should such exist for changed files).
    - [ ] BE: Swagger documentation is updated.
8. [ ] When required by QA:
    - [ ] Deployed to the relevant environment.
    - [ ] Passed system tests.

---

(1) Developer(s) in coordination with QA decide whether it's required. For small tickets introducing small changes QA assistance is most probably not required.

(2) [Octant Areas & Test Cases](https://docs.google.com/spreadsheets/d/1cRe6dxuKJV3a4ZskAwWEPvrFkQm6rEfyUCYwLTYw_Cc).

### CODE REVIEW

**Review by aziolek — COMMENTED** [review]
There are multiple places where BE throws 4** errors that are expected, e.g. `GET /rewards/budget/{user_address}/epoch/{epoch}`. Will those be excluded too?

File: backend/startup.py [hunk 1]
```diff
@@ -20,12 +20,24 @@
 if os.getenv("SENTRY_DSN"):
     import sentry_sdk
 
+    def sentry_before_send(event, hint):
```
Comment (kgarbacinski) [inline backend/startup.py:23]:
```suggestion
    def sentry_before_send(event, hint):
        exceptions = event.get("exception", [])
        if not exceptions:
            return event

        exc = exceptions[-1]
        mechanism = exc.get("mechanism", {})

        if mechanism.get("handled"):
            return None

        return event
```

File: backend/app/infrastructure/apscheduler.py [hunk 2]
```diff
@@ -16,7 +17,10 @@ def vault_confirm_withdrawals():
     with scheduler.app.app_context():
         if app.config["VAULT_CONFIRM_WITHDRAWALS_ENABLED"]:
             app.logger.debug("Confirming withdrawals in Vault contract...")
-            vault.confirm_withdrawals()
+            try:
+                vault.confirm_withdrawals()
+            except exceptions.MissingSnapshot:
+                app.logger.warning("No pending snapshot found")
```
Comment (kgarbacinski) [inline backend/app/infrastructure/apscheduler.py:23]: Is this message correct? (not sure if it's pending snapshot or anything else?)

**Comment by mike-code** [comment]
> There are multiple places where BE throws 4** errors that are expected, e.g. `GET /rewards/budget/{user_address}/epoch/{epoch}`. Will those be excluded too?

if they are handled, then yes

**Review by aziolek — COMMENTED** [review]
There are no changes in this PR.

**Review by paulperegud — APPROVED** [review]

**Pull request merged**